webpack/response_set.js

```diff
@@ -1,27 +1,6 @@
-const $ = require('jquery');
+import React from 'react';
+import ReactDOM from 'react-dom';
+import ResponseSetTable from './components/ResponseSetTable';
 
-exports.addResponseRow = function() {
-  const tableRowCount = $("#response-set-table tr").length - 1;
-  const newRow = `
-        <tr>
-          <td>
-            <label class="hidden" for="response_set_responses_attributes_${tableRowCount}_value">Value</label>
-            <input type="text" name="response_set[responses_attributes][${tableRowCount}][value]" id="response_set_responses_attributes_${tableRowCount}_value">
-          </td>
-          <td>
-            <label class="hidden" for="response_set_responses_attributes_${tableRowCount}_code_system">Code system</label>
-            <input type="text" name="response_set[responses_attributes][${tableRowCount}][code_system]" id="response_set_responses_attributes_${tableRowCount}_code_system">
-          </td>
-          <td>
-            <label class="hidden" for="response_set_responses_attributes_${tableRowCount}_display_name">Display name</label>
-            <input type="text" name="response_set[responses_attributes][${tableRowCount}][display_name]" id="response_set_responses_attributes_${tableRowCount}_display_name">
-          </td>
-      </tr>`;
-  const tbl = $('#response-set-table > tbody:last-child');
-  tbl.append(newRow);
-}
-
-exports.removeResponseRow = function(event) {
-  const row = $(event.target).parents("tr");
-  row.remove();
-}
+const responses = JSON.parse(document.getElementById('responses-json').innerHTML);
```

Comment:
@eedrummer is mostly correct. It has to do with how the browser handles the `script` tag. Under normal operation `script` tags are synchronous meaning if the browser sees one it stops parsing the rest of the HTML until the request for the script tag completes. This means for large scripts it can block the initial page render. The approach most folks took/take is to put the `script` tag at the bottom so that the rest of the HTML gets parsed and rendered first then the JS code gets loaded and can muck up the DOM. There's an alternate approach that involves adding an `async` attribute to the `script` tag which will change the browser behavior. I haven't actually ever used this approach, though Google does seem to recommend it https://developers.google.com/speed/docs/insights/BlockingJS

As for why the `#responses-json` element isn't put in the initial script tag, it's so that it's easier to pull out and parse. Though an alternative could be to assign it to a global variable then reference it but that feels ugly.